evm/vm/forks/sharding/vmc_handler.py

```diff
@@ -266,45 +253,58 @@ def get_eligible_proposer(self, shard_id, period=None, gas=None):
         address_in_hex = self.call(tx_detail).get_eligible_proposer(shard_id, period)
         return decode_hex(address_in_hex)
 
-    def deposit(self,
-                validation_code_addr,
-                return_addr,
-                gas=None,
-                gas_price=None):
-        """deposit(validation_code_addr: address, return_addr: address) -> num
+    def deposit(self, gas=None, gas_price=None):
+        """Do deposit to become a validator
         """
         tx_hash = self.send_transaction(
             'deposit',
-            [
-                to_checksum_address(validation_code_addr),
-                to_checksum_address(return_addr),
-            ],
+            [],
             value=self.config['DEPOSIT_SIZE'],
             gas=gas,
             gas_price=gas_price,
         )
         return tx_hash
 
-    def withdraw(self, validator_index, sig, gas=None, gas_price=None):
-        """withdraw(validator_index: num, sig: bytes <= 1000) -> bool
+    def withdraw(self, validator_index, gas=None, gas_price=None):
+        """Withdraw the validator whose index is `validator_index`
         """
         tx_hash = self.send_transaction(
             'withdraw',
             [
                 validator_index,
-                sig,
             ],
             gas=gas,
             gas_price=gas_price,
         )
         return tx_hash
 
-    def add_header(self, header, gas=None, gas_price=None):
-        """add_header(header: bytes <= 4096) -> bool
+    def add_header(self,
```

Comment:
`VMC.add_header` takes a lot of parameters and combines `add_header` inputs and gas setting in the same level. What do you think about pass `CollationHeader` to this function and move logic of `add_header_send_tx` to here so that the params would be clearer:
```python
def add_header(self, collation_header, gas=None, gas_price=None)
```

Reply:
Good idea! I will change it. Thank you.